app.py

```diff
@@ -0,0 +1,112 @@
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+from dataclasses import dataclass
+from args import get_args
+import typing
+import requests
+import json
+import uvicorn
+import yaml
+import threading
+import time
+
+args = get_args()
+with open(args.config, "r") as stream:
+    try:
+        data = yaml.safe_load(stream)
+        API_KEY = data.get("api_key", None)
+        stops = data.get("stops", {})
+    except yaml.YAMLError as exc:
+        print(exc)
+
+BASE_URL = 'https://api.511.org/transit'
+
+app = FastAPI()
+
+origins = ["*"]
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+@dataclass
+class Prediction:
+  route: str
+  times: typing.List[str]
+
+@dataclass
+class Stop:
+  id: str
+  name: str
+  predictions: typing.List[Prediction]
+
+def update_cache():
+    # clear cache of any previous (now outdated) data
+    cache.clear()
+    # send post request for each agency's stop(s)
+    for stop in stops:
+        agency = stop['operator']
+        stopCode = stop['id']
+        stopName= stop['name']
+
+        params = {
+            'api_key': API_KEY,
+            'agency': agency,
+            'stopCode': stopCode,
+            'format': 'json'
+        }
+
+        response = requests.get(BASE_URL + '/StopMonitoring', params=params)
+
+        if response.status_code == 200:
```

Comment:
instead lets do
```suggestion
        if response.status_code != 200:
            logging.error(f"not parsing response because response code was {response.status_code}")
        # existing code here
```